src/Controls/src/Core/Handlers/Items/Android/RecyclerViewScrollListener.cs

```diff
@@ -34,12 +34,9 @@ public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
 		{
 			base.OnScrolled(recyclerView, dx, dy);
 
-			// TODO: These offsets will be incorrect upon row size or count change.
-			// They are currently provided in place of LayoutManager's default offset calculation
-			// because it does not report accurate values in the presence of uneven rows.
-			// See https://stackoverflow.com/questions/27507715/android-how-to-get-the-current-x-offset-of-recyclerview
-			_horizontalOffset += dx;
-			_verticalOffset += dy;
+			var itemCount = recyclerView.GetAdapter()?.ItemCount;
```

Comment:
`var itemCount = recyclerView.GetAdapter()?.ItemCount ?? 0;`

Just ensure `itemCount` defaults to 0 if the adapter is null.

Reply:
@jsuarezruiz, I have handled null adapter case by defaulting itemcount to 0. Let me know if further changes are needed